nixos/modules/services/backup/restic.nix

```diff
@@ -89,6 +89,34 @@ with lib;
             Create the repository if it doesn't exist.
           '';
         };
+
+        pruneOpts = mkOption {
+          type = types.listOf types.str;
+          default = [];
+          description = ''
+            A list of options (--keep-* et al.) for 'restic forget
+            --prune', to automatically prune old snapshots.  The
+            'forget' command is run *after* the 'backup' command, so
+            keep that in mind when constructing the --keep-* options.
+          '';
+          example = [
+            "--keep-daily 7"
+            "--keep-weekly 5"
+            "--keep-monthly 12"
+            "--keep-yearly 75"
+          ];
+        };
+
+        dynamicFilesFromCmd = mkOption {
+          type = with types; nullOr str;
+          default = null;
+          description = ''
+            A command that produces a list of files to back up.  The
+            results of this command are piped to a temporary file,
+            which is then given to the '--files-from' option.
+          '';
+          example = "find /home/matt/git -type d -name .git";
```

Comment:
This example could be covered by `--exclude *.git`. Other users might have more creative use cases though.

Reply:
I don't want to _exclude_ the `.git/` folders, I want to _include only those folders_.  (My thinking is, if it wasn't important enough to warrant a commit or a stash, it's not worth backing up.)

Reply:
ok. So you want the `--include` flag then?

Reply:
I don't see any `--include` flag documented in `restic help backup`.  (I was surprised not to ...)